File: bundle/wincompat.py
"""Windows 10+ runtime checks for packaged Ledgerly apps."""

from __future__ import annotations

import sys


def require_windows_10() -> None:
    if sys.platform != "win32":
        return
    version = sys.getwindowsversion()
    if version.major < 10:
        raise SystemExit(
            "Ledgerly requires Windows 10 or later.\n"
            f"Detected Windows version: {version.major}.{version.minor}"
        )

File: src/Ledgerly.Server/WinCompat.cs
using System;

namespace Ledgerly.Server;

public static class WinCompat
{
    public static void RequireWindows7OrLater()
    {
        var v = Environment.OSVersion.Version;
        // Windows 7 is 6.1
        if (v.Major < 6 || (v.Major == 6 && v.Minor < 1))
        {
            throw new PlatformNotSupportedException(
                $"Ledgerly requires Windows 7 SP1 or later. Detected {Environment.OSVersion.VersionString}.");
        }
    }
}
